app/models/enrollment.rb

```diff
@@ -1,3 +1,18 @@
 class Enrollment < ApplicationRecord
+  include Cachable
+
   belongs_to :event
+
+  validate :event_capacity_not_exceeded
+
+  # Force association reload to get the updated count
+  after_commit :reload_event
```

Comment:
Se comentar essa linha, os testes continuam passando. Não ficou claro qual sua necessidade

Reply:
AR força o reload da entidade event. Nos testes precisei usar o @event.reload pois no escopo não contempla a mesma transação. Mas irei mudar no mesmo refactoring da camada da transaction direta no model, conforme sugerido